app-android/src/main/kotlin/com/ustadmobile/port/android/view/RoleEditFragment.kt

```diff
@@ -0,0 +1,168 @@
+package com.ustadmobile.port.android.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Observer
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.toughra.ustadmobile.R
+import com.toughra.ustadmobile.databinding.ActivityRoleEditBinding
+import com.toughra.ustadmobile.databinding.ItemBitmaskBinding
+import com.ustadmobile.core.controller.RoleEditPresenter
+import com.ustadmobile.core.controller.UstadEditPresenter
+import com.ustadmobile.core.controller.UstadSingleEntityPresenter
+import com.ustadmobile.core.impl.UmAccountManager
+import com.ustadmobile.core.impl.UstadMobileSystemImpl
+import com.ustadmobile.core.model.BitmaskFlag
+import com.ustadmobile.core.util.ext.toNullableStringMap
+import com.ustadmobile.core.util.ext.toStringMap
+import com.ustadmobile.core.view.RoleEditView
+import com.ustadmobile.lib.db.entities.Role
+import com.ustadmobile.port.android.view.ext.setEditFragmentTitle
+import com.ustadmobile.port.android.view.util.CrudEditActivityResultContract
+
+
+fun ComponentActivity.prepareRoleEditCall(callback: (List<Role>?) -> Unit) =
```

Comment:
Prepare call / activity result contract stuff is no longer used and should be removed.